// nextra-theme-docs/dist/components/navbar/index.client.js
"use client";
import { Fragment, jsx, jsxs } from "react/jsx-runtime";
import { c as _c } from "react-compiler-runtime";
import { MenuItem as _MenuItem, Menu, MenuButton, MenuItems } from "@headlessui/react";
import cn from "clsx";
import { Anchor, Button } from "nextra/components";
import { useFSRoute } from "nextra/hooks";
import { ArrowRightIcon, MenuIcon } from "nextra/icons";
import { setMenu, useConfig, useMenu, useThemeConfig } from "../../stores";
const classes = {
  link: cn("x:text-sm x:contrast-more:text-gray-700 x:contrast-more:dark:text-gray-100 x:whitespace-nowrap", "x:text-gray-600 x:hover:text-gray-800 x:dark:text-gray-400 x:dark:hover:text-gray-200", "x:ring-inset")
};
const NavbarMenu = (t0) => {
  const $ = _c(18);
  const {
    menu,
    children
  } = t0;
  let t1;
  if ($[0] !== menu.children) {
    t1 = Object.fromEntries((menu.children || []).map(_temp));
    $[0] = menu.children;
    $[1] = t1;
  } else {
    t1 = $[1];
  }
  const routes = t1;
  let t2;
  if ($[2] === Symbol.for("react.memo_cache_sentinel")) {
    t2 = /* @__PURE__ */ jsx(ArrowRightIcon, { height: "14", className: "x:*:origin-center x:*:transition-transform x:*:rotate-90" });
    $[2] = t2;
  } else {
    t2 = $[2];
  }
  let t3;
  if ($[3] !== children) {
    t3 = /* @__PURE__ */ jsxs(MenuButton, { className: _temp2, children: [
      children,
      t2
    ] });
    $[3] = children;
    $[4] = t3;
  } else {
    t3 = $[4];
  }
  let t4;
  if ($[5] === Symbol.for("react.memo_cache_sentinel")) {
    t4 = {
      to: "top end",
      gap: 10,
      padding: 16
    };
    $[5] = t4;
  } else {
    t4 = $[5];
  }
  let t5;
  if ($[6] !== menu.items || $[7] !== routes) {
    let t62;
    if ($[9] !== menu.items) {
      t62 = menu.items || {};
      $[9] = menu.items;
      $[10] = t62;
    } else {
      t62 = $[10];
    }
    let t72;
    if ($[11] !== routes) {
      t72 = (t8) => {
        const [key, item] = t8;
        return /* @__PURE__ */ jsx(_MenuItem, { as: Anchor, href: item.href || routes[key]?.route, className: _temp4, children: item.title }, key);
      };
      $[11] = routes;
      $[12] = t72;
    } else {
      t72 = $[12];
    }
    t5 = Object.entries(t62).map(t72);
    $[6] = menu.items;
    $[7] = routes;
    $[8] = t5;
  } else {
    t5 = $[8];
  }
  let t6;
  if ($[13] !== t5) {
    t6 = /* @__PURE__ */ jsx(MenuItems, { transition: true, className: _temp3, anchor: t4, children: t5 });
    $[13] = t5;
    $[14] = t6;
  } else {
    t6 = $[14];
  }
  let t7;
  if ($[15] !== t3 || $[16] !== t6) {
    t7 = /* @__PURE__ */ jsxs(Menu, { children: [
      t3,
      t6
    ] });
    $[15] = t3;
    $[16] = t6;
    $[17] = t7;
  } else {
    t7 = $[17];
  }
  return t7;
};
const isMenu = (page) => page.type === "menu";
const ClientNavbar = (t0) => {
  const $ = _c(18);
  const {
    children
  } = t0;
  const items = useConfig().normalizePagesResult.topLevelNavbarItems;
  const themeConfig = useThemeConfig();
  const pathname = useFSRoute();
  const menu = useMenu();
  let t1;
  if ($[0] !== items || $[1] !== pathname) {
    let t22;
    if ($[3] !== pathname) {
      t22 = (page) => {
        if ("display" in page && page.display === "hidden") {
          return;
        }
        if (isMenu(page)) {
          return /* @__PURE__ */ jsx(NavbarMenu, { menu: page, children: page.title }, page.name);
        }
        let href = page.href || page.route || "#";
        if (page.children) {
          href = ("frontMatter" in page ? page.route : page.firstChildRoute) || href;
        }
        const isCurrentPage = page.route === pathname || pathname.startsWith(page.route + "/") || void 0;
        return /* @__PURE__ */ jsx(Anchor, { href, className: cn(classes.link, "x:aria-[current]:font-medium x:aria-[current]:subpixel-antialiased x:aria-[current]:text-current"), "aria-current": isCurrentPage, children: page.title }, href);
      };
      $[3] = pathname;
      $[4] = t22;
    } else {
      t22 = $[4];
    }
    t1 = items.map(t22);
    $[0] = items;
    $[1] = pathname;
    $[2] = t1;
  } else {
    t1 = $[2];
  }
  let t2;
  if ($[5] !== t1) {
    t2 = /* @__PURE__ */ jsx("div", { className: "x:flex x:gap-4 x:overflow-x-auto nextra-scrollbar x:py-1.5 x:max-md:hidden", children: t1 });
    $[5] = t1;
    $[6] = t2;
  } else {
    t2 = $[6];
  }
  let t3;
  if ($[7] !== themeConfig.search) {
    t3 = themeConfig.search && /* @__PURE__ */ jsx("div", { className: "x:max-md:hidden", children: themeConfig.search });
    $[7] = themeConfig.search;
    $[8] = t3;
  } else {
    t3 = $[8];
  }
  let t4;
  if ($[9] !== menu) {
    t4 = cn({
      open: menu
    });
    $[9] = menu;
    $[10] = t4;
  } else {
    t4 = $[10];
  }
  let t5;
  if ($[11] !== t4) {
    t5 = /* @__PURE__ */ jsx(Button, { "aria-label": "Menu", className: _temp5, onClick: _temp7, children: /* @__PURE__ */ jsx(MenuIcon, { height: "24", className: t4 }) });
    $[11] = t4;
    $[12] = t5;
  } else {
    t5 = $[12];
  }
  let t6;
  if ($[13] !== children || $[14] !== t2 || $[15] !== t3 || $[16] !== t5) {
    t6 = /* @__PURE__ */ jsxs(Fragment, { children: [
      t2,
      t3,
      children,
      t5
    ] });
    $[13] = children;
    $[14] = t2;
    $[15] = t3;
    $[16] = t5;
    $[17] = t6;
  } else {
    t6 = $[17];
  }
  return t6;
};
function _temp(route) {
  return [route.name, route];
}
function _temp2(t0) {
  const {
    focus
  } = t0;
  return cn(classes.link, "x:items-center x:flex x:gap-1.5", focus && "x:nextra-focus");
}
function _temp3(t0) {
  const {
    open
  } = t0;
  return cn("x:focus-visible:nextra-focus", open ? "x:opacity-100" : "x:opacity-0", "nextra-scrollbar x:transition-opacity x:motion-reduce:transition-none", "x:border x:border-black/5 x:dark:border-white/20", "x:backdrop-blur-md x:bg-nextra-bg/70", "x:z-20 x:rounded-md x:py-1 x:text-sm x:shadow-lg", "x:max-h-[min(calc(100vh-5rem),256px)]!");
}
function _temp4(t0) {
  const {
    focus: focus_0
  } = t0;
  return cn("x:block x:py-1.5 x:transition-colors x:ps-3 x:pe-9", focus_0 ? "x:text-gray-900 x:dark:text-gray-100" : "x:text-gray-600 x:dark:text-gray-400");
}
function _temp5(t0) {
  const {
    active
  } = t0;
  return cn("nextra-hamburger x:md:hidden", active && "x:bg-gray-400/20");
}
function _temp6(prev) {
  return !prev;
}
function _temp7() {
  return setMenu(_temp6);
}
export {
  ClientNavbar
};
