// nextra/dist/client/components/collapse.js
"use client";
import { jsx } from "react/jsx-runtime";
import { c as _c } from "react-compiler-runtime";
import cn from "clsx";
import { Children, useEffect, useRef, useState } from "react";
const Collapse = (t0) => {
  const $ = _c(21);
  const {
    children,
    isOpen,
    horizontal: t1,
    openDuration: t2,
    closeDuration: t3
  } = t0;
  const horizontal = t1 === void 0 ? false : t1;
  const openDuration = t2 === void 0 ? 500 : t2;
  const closeDuration = t3 === void 0 ? 300 : t3;
  const containerRef = useRef(null);
  const [initialOpen] = useState(isOpen);
  const animationRef = useRef(0);
  const initialRender = useRef(true);
  let t4;
  let t5;
  if ($[0] !== horizontal || $[1] !== isOpen || $[2] !== openDuration) {
    t4 = () => {
      const animation = animationRef.current;
      const container = containerRef.current;
      if (animation) {
        clearTimeout(animation);
        animationRef.current = 0;
      }
      if (initialRender.current) {
        return;
      }
      const child = container.children[0];
      if (horizontal) {
        child.style.width = `${child.clientWidth}px`;
        container.style.width = `${child.clientWidth}px`;
      } else {
        container.style.height = `${child.clientHeight}px`;
      }
      if (isOpen) {
        animationRef.current = window.setTimeout(() => {
          container.style.removeProperty("height");
        }, openDuration);
      } else {
        setTimeout(() => {
          if (horizontal) {
            container.style.width = "0";
          } else {
            container.style.height = "0";
          }
        });
      }
    };
    t5 = [horizontal, isOpen, openDuration];
    $[0] = horizontal;
    $[1] = isOpen;
    $[2] = openDuration;
    $[3] = t4;
    $[4] = t5;
  } else {
    t4 = $[3];
    t5 = $[4];
  }
  useEffect(t4, t5);
  let t6;
  let t7;
  if ($[5] === Symbol.for("react.memo_cache_sentinel")) {
    t6 = () => {
      initialRender.current = false;
    };
    t7 = [];
    $[5] = t6;
    $[6] = t7;
  } else {
    t6 = $[5];
    t7 = $[6];
  }
  useEffect(t6, t7);
  let t8;
  if ($[7] !== children) {
    t8 = Children.count(children) === 1 && children && typeof children === "object" && "type" in children ? children : /* @__PURE__ */ jsx("div", { children });
    $[7] = children;
    $[8] = t8;
  } else {
    t8 = $[8];
  }
  const newChildren = t8;
  let t9;
  if ($[9] !== isOpen) {
    t9 = cn("x:transform-gpu x:transition-all x:ease-in-out x:motion-reduce:transition-none", isOpen ? "x:opacity-100" : ["x:opacity-0", "x:overflow-hidden"]);
    $[9] = isOpen;
    $[10] = t9;
  } else {
    t9 = $[10];
  }
  let t10;
  if ($[11] !== horizontal || $[12] !== initialOpen) {
    t10 = initialOpen || horizontal ? void 0 : {
      height: 0
    };
    $[11] = horizontal;
    $[12] = initialOpen;
    $[13] = t10;
  } else {
    t10 = $[13];
  }
  const t11 = (isOpen ? openDuration : closeDuration) + "ms";
  let t12;
  if ($[14] !== t10 || $[15] !== t11) {
    t12 = {
      ...t10,
      transitionDuration: t11
    };
    $[14] = t10;
    $[15] = t11;
    $[16] = t12;
  } else {
    t12 = $[16];
  }
  let t13;
  if ($[17] !== newChildren || $[18] !== t12 || $[19] !== t9) {
    t13 = /* @__PURE__ */ jsx("div", { ref: containerRef, className: t9, style: t12, children: newChildren });
    $[17] = newChildren;
    $[18] = t12;
    $[19] = t9;
    $[20] = t13;
  } else {
    t13 = $[20];
  }
  return t13;
};
export {
  Collapse
};
