// nextra-theme-docs/dist/components/toc.js
"use client";
import { Fragment, jsx, jsxs } from "react/jsx-runtime";
import { c as _c } from "react-compiler-runtime";
import cn from "clsx";
import { Anchor } from "nextra/components";
import { useEffect, useRef } from "react";
import scrollIntoView from "scroll-into-view-if-needed";
import { useActiveAnchor, useThemeConfig } from "../stores";
import { getGitIssueUrl, gitUrlParse } from "../utils";
import { BackToTop } from "./back-to-top";
const linkClassName = cn("x:text-xs x:font-medium", "x:text-gray-600 x:dark:text-gray-400", "x:hover:text-gray-800 x:dark:hover:text-gray-200", "x:contrast-more:text-gray-700 x:contrast-more:dark:text-gray-100");
const TOC = (t0) => {
  const $ = _c(25);
  const {
    toc,
    filePath,
    pageTitle
  } = t0;
  const activeSlug = useActiveAnchor();
  const tocRef = useRef(null);
  const themeConfig = useThemeConfig();
  const hasMetaInfo = themeConfig.feedback.content || themeConfig.editLink || themeConfig.toc.extraContent || themeConfig.toc.backToTop;
  const hasHeadings = toc.length > 0;
  let t1;
  if ($[0] !== activeSlug) {
    t1 = (t22) => {
      const {
        id
      } = t22;
      return id === activeSlug;
    };
    $[0] = activeSlug;
    $[1] = t1;
  } else {
    t1 = $[1];
  }
  const activeIndex = toc.findIndex(t1);
  let t2;
  let t3;
  if ($[2] !== activeSlug) {
    t2 = () => {
      if (!activeSlug) {
        return;
      }
      const anchor = tocRef.current?.querySelector(`a[href="#${activeSlug}"]`);
      if (!anchor) {
        return;
      }
      scrollIntoView(anchor, {
        behavior: "smooth",
        block: "center",
        inline: "center",
        scrollMode: "if-needed",
        boundary: tocRef.current
      });
    };
    t3 = [activeSlug];
    $[2] = activeSlug;
    $[3] = t2;
    $[4] = t3;
  } else {
    t2 = $[3];
    t3 = $[4];
  }
  useEffect(t2, t3);
  let t4;
  if ($[5] === Symbol.for("react.memo_cache_sentinel")) {
    t4 = cn("x:grid x:grid-rows-[min-content_1fr_min-content]", "x:sticky x:top-(--nextra-navbar-height) x:text-sm", "x:max-h-[calc(100vh-var(--nextra-navbar-height))]");
    $[5] = t4;
  } else {
    t4 = $[5];
  }
  let t5;
  if ($[6] !== activeSlug || $[7] !== hasHeadings || $[8] !== themeConfig.toc || $[9] !== toc) {
    t5 = hasHeadings && /* @__PURE__ */ jsxs(Fragment, { children: [
      /* @__PURE__ */ jsx("p", { className: "x:pt-6 x:px-4 x:font-semibold x:tracking-tight", children: themeConfig.toc.title }),
      /* @__PURE__ */ jsx("ul", { ref: tocRef, className: cn("x:p-4 nextra-scrollbar x:overscroll-y-contain x:overflow-y-auto x:hyphens-auto", "nextra-mask"), children: toc.map((t62) => {
        const {
          id: id_0,
          value,
          depth
        } = t62;
        return /* @__PURE__ */ jsx("li", { className: "x:my-2 x:scroll-my-6 x:scroll-py-6", children: /* @__PURE__ */ jsx("a", { href: `#${id_0}`, className: cn("x:focus-visible:nextra-focus", {
          2: "x:font-semibold",
          3: "x:ms-3",
          4: "x:ms-6",
          5: "x:ms-9",
          6: "x:ms-12"
        }[depth], "x:block x:transition-colors x:subpixel-antialiased", id_0 === activeSlug ? "x:text-primary-600 x:contrast-more:text-primary-600!" : "x:text-gray-500 x:hover:text-gray-900 x:dark:text-gray-400 x:dark:hover:text-gray-300", "x:contrast-more:text-gray-900 x:contrast-more:underline x:contrast-more:dark:text-gray-50 x:break-words"), children: value }) }, id_0);
      }) })
    ] });
    $[6] = activeSlug;
    $[7] = hasHeadings;
    $[8] = themeConfig.toc;
    $[9] = toc;
    $[10] = t5;
  } else {
    t5 = $[10];
  }
  let t6;
  if ($[11] !== activeIndex || $[12] !== filePath || $[13] !== hasHeadings || $[14] !== hasMetaInfo || $[15] !== pageTitle || $[16] !== themeConfig.docsRepositoryBase || $[17] !== themeConfig.editLink || $[18] !== themeConfig.feedback.content || $[19] !== themeConfig.feedback.labels || $[20] !== themeConfig.toc) {
    t6 = hasMetaInfo && /* @__PURE__ */ jsxs("div", { className: cn("x:grid x:gap-2 x:py-4 x:mx-4", hasHeadings && "x:border-t nextra-border"), children: [
      themeConfig.feedback.content && /* @__PURE__ */ jsx(Anchor, { className: linkClassName, href: getGitIssueUrl({
        labels: themeConfig.feedback.labels,
        repository: themeConfig.docsRepositoryBase,
        title: `Feedback for \u201C${pageTitle}\u201D`
      }), children: themeConfig.feedback.content }),
      filePath && themeConfig.editLink && /* @__PURE__ */ jsx(Anchor, { className: linkClassName, href: filePath.startsWith("http") ? filePath : `${gitUrlParse(themeConfig.docsRepositoryBase).href}/${filePath}`, children: themeConfig.editLink }),
      themeConfig.toc.extraContent,
      themeConfig.toc.backToTop && /* @__PURE__ */ jsx(BackToTop, { className: linkClassName, hidden: activeIndex < 2, children: themeConfig.toc.backToTop })
    ] });
    $[11] = activeIndex;
    $[12] = filePath;
    $[13] = hasHeadings;
    $[14] = hasMetaInfo;
    $[15] = pageTitle;
    $[16] = themeConfig.docsRepositoryBase;
    $[17] = themeConfig.editLink;
    $[18] = themeConfig.feedback.content;
    $[19] = themeConfig.feedback.labels;
    $[20] = themeConfig.toc;
    $[21] = t6;
  } else {
    t6 = $[21];
  }
  let t7;
  if ($[22] !== t5 || $[23] !== t6) {
    t7 = /* @__PURE__ */ jsxs("div", { className: t4, children: [
      t5,
      t6
    ] });
    $[22] = t5;
    $[23] = t6;
    $[24] = t7;
  } else {
    t7 = $[24];
  }
  return t7;
};
export {
  TOC
};
